enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/compiled/codegen/CodeGenerator.scala

```diff
@@ -72,11 +73,11 @@ class CodeGenerator(val structure: CodeStructure[GeneratedQuery], clock: Clock,
 
         val builder = new RunnablePlan {
           def apply(queryContext: QueryContext, execMode: ExecutionMode,
-                    descriptionProvider: DescriptionProvider, params: Map[String, Any],
+                    descriptionProvider: DescriptionProvider, params: MapValue,
                     closer: TaskCloser): InternalExecutionResult = {
             val (provider, tracer) = descriptionProvider(description)
             val execution: GeneratedQueryExecution = query.query.execute(closer, queryContext, execMode,
-              provider, tracer.getOrElse(QueryExecutionTracer.NONE), asJavaHashMap(params))
```

Comment:
This is where the money is, no?

Reply:
Nah, this is saves us one copy. The bigger thing is that we before first turned `Map<String, Object>` into `MapValue` and then the compiled runtime took the `MapValue` and turned that into a `Map<String,Object>` again and then before actually using it in the compiled runtime it passes `CompiledConversionUtils.loadParameter` which introspects and transforms the values in the map again.